tests/Unleash.Tests/Internal/Dependent_Features_Tests.cs

```diff
@@ -0,0 +1,610 @@
+
+            warnOnce.Warn("test2", "testmessage2");
+            warnOnce.Warn("test2", "testmessage2");
+
+            // Assert
+            A.CallTo(() => logger.Log(A<LogLevel>._, A<Func<string>>._, null)).MustHaveHappenedTwiceExactly();
+        }
+
+        [Test]
+        public void Depends_On_One_Enabled_Parent_IsEnabled_True()
+        {
+            // Arrange
+            var appname = "testapp";
+            var dependencies = new List<Dependency>()
+            {
+                new Dependency("parent-enabled-1"),
+            };
+            var toggles = new List<FeatureToggle>()
+            {
+                ParentEnabledOne(),
+                ParentEnabledTwo(),
+                ChildDependentOn("child-1", dependencies)
+            };
+            var state = new ToggleCollection(toggles);
+            state.Version = 2;
+            var unleash = CreateUnleash(appname, state);
+
+            // Act
+            var result = unleash.IsEnabled("child-1");
+
+            // Assert
+            result.Should().BeTrue();
+        }
+
+        [Test]
+        public void Depends_On_One_Enabled_Parent_With_Variants_Red_Or_Blue_IsEnabled_True()
+        {
+            // Arrange
+            var appname = "testapp";
+            var dependencies = new List<Dependency>()
+            {
+                new Dependency("parent-variants-enabled-1", variants: new [] { "red", "blue" }),
+            };
+            var toggles = new List<FeatureToggle>()
+            {
+                ParentWithVariantsRedBlueEnabledOne(),
+                ParentEnabledTwo(),
+                ChildDependentOn("child-1", dependencies)
+            };
+            var state = new ToggleCollection(toggles);
+            state.Version = 2;
+            var unleash = CreateUnleash(appname, state);
+
+            // Act
+            var result = unleash.IsEnabled("child-1");
+
+            // Assert
+            result.Should().BeTrue();
+        }
+
+        [Test]
+        public void Depends_On_One_Enabled_Parent_Counts_Metrics_Only_For_Child()
+        {
+            // Arrange
+            var appname = "testapp";
+            var dependencies = new List<Dependency>()
+            {
+                new Dependency("parent-enabled-1"),
+            };
+            var toggles = new List<FeatureToggle>()
+            {
+                ParentEnabledOne(),
+                ParentEnabledTwo(),
+                ChildDependentOn("child-1", dependencies)
+            };
+            var state = new ToggleCollection(toggles);
+            state.Version = 2;
+            var unleash = CreateUnleash(appname, state);
+
+            // Act
+            var result = unleash.IsEnabled("child-1");
+
+            // Assert
+            unleash.services.MetricsBucket.StopCollectingMetrics(out var bucket);
+            var childMetrics = bucket.Toggles.Single(t => t.Key == "child-1").Value;
+            (childMetrics.No + childMetrics.Yes).Should().Be(1L);
```

Comment:
personally I prefer explicit numbers in tests instead of calculations

Reply:
Fair point, the explicit number was in the .Be part of the assertion - but I'll add a separate line for each to not sum the metrics no and yes!